upgrades/backend.go

```diff
@@ -289,153 +156,14 @@ func (s stateBackend) LegacyLeases(localTime time.Time) (map[lease.Key]lease.Inf
 	return state.LegacyLeases(s.pool, localTime)
 }
 
-func (s stateBackend) SetEnableDiskUUIDOnVsphere() error {
-	return state.SetEnableDiskUUIDOnVsphere(s.pool)
-}
-
-func (s stateBackend) UpdateInheritedControllerConfig() error {
-	return state.UpdateInheritedControllerConfig(s.pool)
-}
-
-func (s stateBackend) UpdateKubernetesStorageConfig() error {
-	return state.UpdateKubernetesStorageConfig(s.pool)
-}
-
-func (s stateBackend) EnsureDefaultModificationStatus() error {
-	return state.EnsureDefaultModificationStatus(s.pool)
-}
-
-func (s stateBackend) EnsureApplicationDeviceConstraints() error {
-	return state.EnsureApplicationDeviceConstraints(s.pool)
-}
-
-func (s stateBackend) RemoveInstanceCharmProfileDataCollection() error {
-	return state.RemoveInstanceCharmProfileDataCollection(s.pool)
-}
-
-func (s stateBackend) UpdateK8sModelNameIndex() error {
-	return state.UpdateK8sModelNameIndex(s.pool)
-}
-
-func (s stateBackend) AddModelLogsSize() error {
-	return state.AddModelLogsSize(s.pool)
-}
-
-func (s stateBackend) AddControllerNodeDocs() error {
-	return state.AddControllerNodeDocs(s.pool)
-}
-
-func (s stateBackend) AddSpaceIdToSpaceDocs() error {
-	return state.AddSpaceIdToSpaceDocs(s.pool)
-}
-
-func (s stateBackend) ChangeSubnetAZtoSlice() error {
-	return state.ChangeSubnetAZtoSlice(s.pool)
-}
-
-func (s stateBackend) ChangeSubnetSpaceNameToSpaceID() error {
-	return state.ChangeSubnetSpaceNameToSpaceID(s.pool)
-}
-
-func (s stateBackend) AddSubnetIdToSubnetDocs() error {
-	return state.AddSubnetIdToSubnetDocs(s.pool)
-}
-
-func (s stateBackend) ReplacePortsDocSubnetIDCIDR() error {
-	return state.ReplacePortsDocSubnetIDCIDR(s.pool)
-}
-
-func (s stateBackend) EnsureRelationApplicationSettings() error {
-	return state.EnsureRelationApplicationSettings(s.pool)
-}
-
-func (s stateBackend) ConvertAddressSpaceIDs() error {
-	return state.ConvertAddressSpaceIDs(s.pool)
-}
-
-func (s stateBackend) ReplaceSpaceNameWithIDEndpointBindings() error {
-	return state.ReplaceSpaceNameWithIDEndpointBindings(s.pool)
-}
-
-func (s stateBackend) EnsureDefaultSpaceSetting() error {
-	return state.EnsureDefaultSpaceSetting(s.pool)
-}
-func (s stateBackend) RemoveControllerConfigMaxLogAgeAndSize() error {
-	return state.RemoveControllerConfigMaxLogAgeAndSize(s.pool)
-}
-
-func (s stateBackend) IncrementTasksSequence() error {
-	return state.IncrementTasksSequence(s.pool)
-}
-
-func (s stateBackend) AddMachineIDToSubordinates() error {
-	return state.AddMachineIDToSubordinates(s.pool)
-}
-
-func (s stateBackend) AddOriginToIPAddresses() error {
-	return state.AddOriginToIPAddresses(s.pool)
-}
-
-func (s stateBackend) DropPresenceDatabase() error {
-	return state.DropPresenceDatabase(s.pool)
-}
-
 func (s stateBackend) DropLeasesCollection() error {
```

Comment:
This can go too. It's used by `MigrateLegacyLeases` in upgrades/raft.go, which can also be deleted.

Reply:
Removed `DropLeasesCollection()` and `MigrateLegacyLeases`from upgrades/raft.go